web_client/dialogs/editRegionOfInterest.js

```diff
@@ -0,0 +1,207 @@
+
+import View from 'girder/views/View';
+import { apiRoot } from 'girder/rest';
+import { formatSize } from 'girder/misc';
+
+import router from '../router';
+import editRegionOfInterest from '../templates/dialogs/editRegionOfInterest.pug';
+import '../stylesheets/panels/zoomWidget.styl';
+
+var EditRegionOfInterest = View.extend({
+    events: {
+        'change .update-form': 'updateform'
+    },
+
+    initialize() {
+        this._sizeCte = 1000;    // Constante to find
+        this._format = 'JPEG';    // JPEG is the default format
+        this._compressionRatio = 0.35;    // JPEG ratio
+        this._magnification = 0;
+    },
+
+    render() {
+        if (this._magnification <= 1) {
+            this._magnification = 1;
+        } else if (this._magnification >= this.areaElement.maxMag) {
+            this._magnification = this.areaElement.maxMag;
+        }
+        var bounds = this.scaleBounds();
+        this.$el.html(
+            editRegionOfInterest({
+                magnification: this._magnification,
+                scaleWidth: bounds.width,
+                scaleHeight: bounds.height,
+                maxMag: this.areaElement.maxMag,
+                numberOfPixel: this.getNumberPixels(),
+                fileSize: this.getConvertFileSize()
+            })
+        ).girderModal(this);
+        this.updateform();
+        return this;
+    },
+
+    /**
+     * Convert from zoom level to magnification.
+     */
+    zoomToMagnification(zoom) {
+        return Math.round(parseFloat(this.areaElement.maxMag) *
+            Math.pow(2, zoom - parseFloat(this.areaElement.maxZoom)) * 10) / 10;
+    },
+
+    /**
+     * Convert from magnification to zoom level.
+     */
+    magnificationToZoom(magnification) {
+        return parseFloat(this.areaElement.maxZoom) -
+            Math.log2(this.areaElement.maxMag / magnification);
+    },
+
+    /**
+     * Convert from magnification to zoom level.
+     */
+    scaleBounds() {
+        var zoom = this.magnificationToZoom(this._magnification);
+        var factor = Math.pow(2, zoom - this.areaElement.maxZoom);
+        var scaleWidth = Math.round(factor * this.areaElement.width);
+        var scaleHeight = Math.round(factor * this.areaElement.height);
+        return { 'width': scaleWidth, 'height': scaleHeight };
+    },
+
+    /**
+     * Get the number of pixel in the region of interest
+     */
+    getNumberPixels() {
+        var bounds = this.scaleBounds();
+        var Npixel = bounds.width * bounds.height;
+        return Npixel;
+    },
+
+    /**
+     * Get the size of the file before download it for an image in 24b/px (result in Bytes)
+     */
+    getFileSize() {
+        var fileSize = (this.getNumberPixels() * 3 + this._sizeCte) * this._compressionRatio;
+        return fileSize;
+    },
+
+    /**
+     * Get the size of the file in the appropriate unity (Bytes, MB, GB...)
+     */
+    getConvertFileSize() {
+        var Nbytes = this.getFileSize();
```

Comment:
Here an elsewhere, avoid making variables names starting with capital letters.  These are usually reserved for classes.